### PR TITLE
Escape percentage symbol in argparse

### LINKED ISSUE #498: Escaping % in add_default_args
**Describe the bug**

In utilities/arg_parse.py, a percentage symbol is not escaped and would cause an error when printing help information.

```python
parser.add_argument('--overfit', default=-1, type=float,
                        help='% of dataset to use with this option. float, or -1 for none')
```

**To Reproduce**
```
import os
import random
import sys
from pytorch_lightning.utilities.arg_parse import add_default_args
from test_tube import HyperOptArgumentParser, Experiment

if __name__ == "__main__":
    root_dir = os.path.split(os.path.dirname(sys.modules['__main__'].__file__))[0]
    parent_parser = HyperOptArgumentParser(strategy='random_search', add_help=True)
    add_default_args(parent_parser, root_dir)
    hyperparams = parent_parser.parse_args()
```
Execute the file with `--help`

```
python temp.py --help
```

Throws an error:
```
WARNING:root:This caffe2 python run does not have GPU support. Will run in CPU only mode.
Traceback (most recent call last):
  File "/Users/chenghaomou/Code/ai2/temp.py", line 11, in <module>
    hyperparams = parent_parser.parse_args()
  File "/Users/chenghaomou/Anaconda/envs/Elisa/lib/python3.7/site-packages/test_tube/argparse_hopt.py", line 238, in parse_args
    results = self.__parse_args(args, namespace)
  File "/Users/chenghaomou/Anaconda/envs/Elisa/lib/python3.7/site-packages/test_tube/argparse_hopt.py", line 157, in __parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "/Users/chenghaomou/Anaconda/envs/Elisa/lib/python3.7/argparse.py", line 1782, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/Users/chenghaomou/Anaconda/envs/Elisa/lib/python3.7/argparse.py", line 1988, in _parse_known_args
    start_index = consume_optional(start_index)
  File "/Users/chenghaomou/Anaconda/envs/Elisa/lib/python3.7/argparse.py", line 1928, in consume_optional
    take_action(action, args, option_string)
  File "/Users/chenghaomou/Anaconda/envs/Elisa/lib/python3.7/argparse.py", line 1856, in take_action
    action(self, namespace, argument_values, option_string)
  File "/Users/chenghaomou/Anaconda/envs/Elisa/lib/python3.7/argparse.py", line 1038, in __call__
    parser.print_help()
  File "/Users/chenghaomou/Anaconda/envs/Elisa/lib/python3.7/argparse.py", line 2475, in print_help
    self._print_message(self.format_help(), file)
  File "/Users/chenghaomou/Anaconda/envs/Elisa/lib/python3.7/argparse.py", line 2459, in format_help
    return formatter.format_help()
  File "/Users/chenghaomou/Anaconda/envs/Elisa/lib/python3.7/argparse.py", line 284, in format_help
    help = self._root_section.format_help()
  File "/Users/chenghaomou/Anaconda/envs/Elisa/lib/python3.7/argparse.py", line 215, in format_help
    item_help = join([func(*args) for func, args in self.items])
  File "/Users/chenghaomou/Anaconda/envs/Elisa/lib/python3.7/argparse.py", line 215, in <listcomp>
    item_help = join([func(*args) for func, args in self.items])
  File "/Users/chenghaomou/Anaconda/envs/Elisa/lib/python3.7/argparse.py", line 215, in format_help
    item_help = join([func(*args) for func, args in self.items])
  File "/Users/chenghaomou/Anaconda/envs/Elisa/lib/python3.7/argparse.py", line 215, in <listcomp>
    item_help = join([func(*args) for func, args in self.items])
  File "/Users/chenghaomou/Anaconda/envs/Elisa/lib/python3.7/argparse.py", line 525, in _format_action
    help_text = self._expand_help(action)
  File "/Users/chenghaomou/Anaconda/envs/Elisa/lib/python3.7/argparse.py", line 615, in _expand_help
    return self._get_help_string(action) % params
TypeError: %o format: an integer is required, not dict
```

**Expected behavior**
Escape the percentage sign and help can be printed. 

**Desktop (please complete the following information):**
 - OS: macOS 10.15
 - Browser Chrome
 - Version 78.0.3904.87

**Additional context**

### PR DESCRIPTION
**Pull request by ChenghaoMou**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/williamFalcon/pytorch-lightning/blob/master/.github/CONTRIBUTING.md)?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  

## What does this PR do?
Fixes #498 (issue).

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Yes

### CODE REVIEW

File: pytorch_lightning/utilities/arg_parse.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def add_default_args(parser, root_dir, rand_seed=None, possible_model_names=None
     parser.add_argument('--enable_tqdm', dest='enable_tqdm', default=False, action='store_true',
                         help='false removes the progress bar')
     parser.add_argument('--overfit', default=-1, type=float,
-                        help='% of dataset to use with this option. float, or -1 for none')
+                        help='%% of dataset to use with this option. float, or -1 for none')
```
Comment (Borda) [inline pytorch_lightning/utilities/arg_parse.py]: "to be used" and pls look to the code and adjust the value range of it is (0, 1) or (0, 100)

**Comment by williamFalcon** [comment]
@Borda is this good to go?

**Comment by Borda** [comment]
@williamFalcon well just see it again and the change is not needed, `%%` is printed as %% since there is no formating... co probably useless change... sorry

**Comment by ChenghaoMou** [comment]
After changing % to %%, I was able to print the help messages. It also displays as %. I am using python version 3.7.

**Comment by Borda** [comment]
@ChenghaoMou so you say that it was not possible to print help before? and what about py3.6?

**Comment by ChenghaoMou** [comment]
@Borda I just created a new conda env with python 3.6 and ran the following code:

```
import os
import random
import sys
from pytorch_lightning.utilities.arg_parse import add_default_args
from test_tube import HyperOptArgumentParser, Experiment

if __name__ == "__main__":
    root_dir = os.path.split(os.path.dirname(sys.modules['__main__'].__file__))[0]
    parent_parser = HyperOptArgumentParser(strategy='random_search', add_help=True)
    add_default_args(parent_parser, root_dir)
    hyperparams = parent_parser.parse_args()
```

It throws the error:

```
Traceback (most recent call last):
  File "test.py", line 11, in <module>
    hyperparams = parent_parser.parse_args()
  File "/Users/chenghaomou/Anaconda/envs/test/lib/python3.6/site-packages/test_tube/argparse_hopt.py", line 237, in parse_args
    results = self.__parse_args(args, namespace)
  File "/Users/chenghaomou/Anaconda/envs/test/lib/python3.6/site-packages/test_tube/argparse_hopt.py", line 157, in __parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "/Users/chenghaomou/Anaconda/envs/test/lib/python3.6/argparse.py", line 1766, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/Users/chenghaomou/Anaconda/envs/test/lib/python3.6/argparse.py", line 1972, in _parse_known_args
    start_index = consume_optional(start_index)
  File "/Users/chenghaomou/Anaconda/envs/test/lib/python3.6/argparse.py", line 1912, in consume_optional
    take_action(action, args, option_string)
  File "/Users/chenghaomou/Anaconda/envs/test/lib/python3.6/argparse.py", line 1840, in take_action
    action(self, namespace, argument_values, option_string)
  File "/Users/chenghaomou/Anaconda/envs/test/lib/python3.6/argparse.py", line 1024, in __call__
    parser.print_help()
  File "/Users/chenghaomou/Anaconda/envs/test/lib/python3.6/argparse.py", line 2366, in print_help
    self._print_message(self.format_help(), file)
  File "/Users/chenghaomou/Anaconda/envs/test/lib/python3.6/argparse.py", line 2350, in format_help
    return formatter.format_help()
  File "/Users/chenghaomou/Anaconda/envs/test/lib/python3.6/argparse.py", line 282, in format_help
    help = self._root_section.format_help()
  File "/Users/chenghaomou/Anaconda/envs/test/lib/python3.6/argparse.py", line 213, in format_help
    item_help = join([func(*args) for func, args in self.items])
  File "/Users/chenghaomou/Anaconda/envs/test/lib/python3.6/argparse.py", line 213, in <listcomp>
    item_help = join([func(*args) for func, args in self.items])
  File "/Users/chenghaomou/Anaconda/envs/test/lib/python3.6/argparse.py", line 213, in format_help
    item_help = join([func(*args) for func, args in self.items])
  File "/Users/chenghaomou/Anaconda/envs/test/lib/python3.6/argparse.py", line 213, in <listcomp>
    item_help = join([func(*args) for func, args in self.items])
  File "/Users/chenghaomou/Anaconda/envs/test/lib/python3.6/argparse.py", line 523, in _format_action
    help_text = self._expand_help(action)
  File "/Users/chenghaomou/Anaconda/envs/test/lib/python3.6/argparse.py", line 610, in _expand_help
    return self._get_help_string(action) % params
TypeError: %o format: an integer is required, not dict
```

It is also mentioned that you have to add % to escape literal % in python documentation: https://docs.python.org/dev/library/argparse.html#help

```
As the help string supports %-formatting, if you want a literal % to appear in the help string, you must escape it as %%.
```

**Comment by ChenghaoMou** [comment]
I started to use lightning about two months ago, I wasn't able to print then and I had to change the percentage.

**Comment by Borda** [comment]
check and you are right, also the change fix it... @williamFalcon ready to go

**Pull request merged**